radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/addressbook/AddressBook.java

```diff
@@ -92,65 +91,33 @@
 
 /** Manages known peers network addresses and their metadata. */
 public final class AddressBook {
-
-  /**
-   * A stateful comparator for known peer addresses that uses both their latest connection status
-   * (persistent) and a number of failed connection attempts (volatile). Failure counts are used to
-   * cycle the addresses for retries. The entries are sorted in the following manner: successful ->
-   * unknown (no connection attempted yet) -> failed (by num of failures)
-   */
-  private static final class PeerAddressEntryComparator implements Comparator<PeerAddressEntry> {
-    private final Map<RadixNodeUri, Integer> failureCounts = new HashMap<>();
-
-    private int toIntValue(PeerAddressEntry peerAddressEntry) {
-      return peerAddressEntry
-          .getLatestConnectionStatus()
-          .map(
-              latestConnectionStatus -> {
-                if (latestConnectionStatus == LatestConnectionStatus.SUCCESS) {
-                  return 1;
-                } else {
-                  return -(1 + failureCounts.getOrDefault(peerAddressEntry.getUri(), 0));
-                }
-              })
-          .orElse(0);
-    }
-
-    @Override
-    public int compare(PeerAddressEntry a, PeerAddressEntry b) {
-      return Integer.compare(toIntValue(b), toIntValue(a));
-    }
-
-    void incFailures(RadixNodeUri uri) {
-      synchronized (this.failureCounts) {
-        final var curr = this.failureCounts.getOrDefault(uri, 0);
-        this.failureCounts.put(uri, curr + 1);
-      }
-    }
-
-    void resetFailures(RadixNodeUri uri) {
-      synchronized (this.failureCounts) {
-        this.failureCounts.remove(uri);
-      }
-    }
-  }
+  private static final Duration INITIAL_POSTPONE = Duration.ofMinutes(5);
+  private static final Duration MAX_POSTPONE = Duration.ofHours(1);
 
   private final RadixNodeUri self;
   private final EventDispatcher<PeerEvent> peerEventDispatcher;
   private final AddressBookPersistence persistence;
+  private final PeerPostponingManager postponingManager;
```

Comment:
Not sure about this name, but can't think of a better one. @dhedey what do you think?